src/v2/api/index.md

```diff
@@ -1576,6 +1576,34 @@ type: api
   // `callback` is fired immediately with current value of `a`
   ```
 
+  Note that with `immediate` option you won't be able to unwatch the given property on the first callback call. 
+
+  ``` js
+  // This will cause an error
+  const unwatch = this.$watch(
```

Comment:
We have a convention to use ES5 only (except for cookbook), which means no `const` or arrow functions.

Reply:
I'd also suggest using `vm` instead of `this` to match the other examples.

Reply:
Fixed 😅